components/Service-Management.tsx

```diff
@@ -0,0 +1,142 @@
+    const [isAddingService, setIsAddingService] = useState(false);
+    const [editingService, setEditingService] = useState<Service | null>(null);
+    const [newService, setNewService] = useState<Partial<Service>>({});
+
+    const services = useQuery(api.services.list);
+    const addService = useMutation(api.services.add);
+    const updateService = useMutation(api.services.update);
+    const deleteService = useMutation(api.services.remove);
+
+    const handleAddService = async () => {
+        try {
+            await addService(newService);
+            setIsAddingService(false);
+            setNewService({});
+            toast({ title: "Success", description: "Service added successfully" });
+        } catch (error) {
+            console.error("Error adding service:", error);
+            toast({ title: "Error", description: "Failed to add service", variant: "destructive" });
+        }
+    };
+
+    const handleUpdateService = async () => {
+        if (!editingService) return;
+        try {
+            await updateService({ id: editingService._id, ...editingService });
+            setEditingService(null);
+            toast({ title: "Success", description: "Service updated successfully" });
+        } catch (error) {
+            console.error("Error updating service:", error);
+            toast({ title: "Error", description: "Failed to update service", variant: "destructive" });
+        }
+    };
+
+    const handleDeleteService = async (id: string) => {
+        try {
+            await deleteService({ id });
+            toast({ title: "Success", description: "Service deleted successfully" });
+        } catch (error) {
+            console.error("Error deleting service:", error);
+            toast({ title: "Error", description: "Failed to delete service", variant: "destructive" });
+        }
+    };
+
+    if (!services) {
+        return <Spinner />;
+    }
+
+    return (
+        <div className="space-y-4">
+            <h1 className="text-2xl font-bold">Service Management</h1>
+            <Button onClick={() => setIsAddingService(true)}>Add New Service</Button>
+            <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4">
+                {services.map((service) => (
+                    <Card key={service._id}>
+                        <CardHeader>
+                            <CardTitle>{service.name}</CardTitle>
+                        </CardHeader>
+                        <CardContent>
+                            <p>Price: ${service.price.toFixed(2)}</p>
+                            <p>Duration: {service.duration} minutes</p>
+                            <div className="mt-2 space-x-2">
+                                <Button variant="outline" size="sm" onClick={() => setEditingService(service)}>Edit</Button>
+                                <Button variant="destructive" size="sm" onClick={() => handleDeleteService(service._id)}>Delete</Button>

```

Comment:
**Suggestion:** To ensure that the service deletion process is intentional and to prevent accidental deletions, consider implementing a confirmation dialog before executing the delete operation. [usability, importance: 9]
```suggestion
                                <Button variant="destructive" size="sm" onClick={() => confirmDelete(service)}>Delete</Button>
```